hash.go

```diff
@@ -349,7 +366,7 @@ func (h *evpHash) BlockSize() int {
 func (h *evpHash) Sum(in []byte) []byte {
 	h.init()
 	out := make([]byte, h.Size(), maxHashSize) // explicit cap to allow stack allocation
-	if C.go_hash_sum(h.ctx, h.ctx2, base(out)) != 1 {
+	if C.go_hash_sum(h.ctx, h.ctx2, (*C.uchar)(unsafe.SliceData(out))) != 1 {
```

Comment:
What is this change for?

Reply:
`base` no longer returns a `*C.char`. This is the only remaining place where we need to cast a slice to `*C.char`, so the case is don't in place.